src/library/Uno.Material/Controls/ExpandingBottomSheet.cs

```diff
@@ -0,0 +1,139 @@
+﻿using Windows.ApplicationModel.Store;
+using Windows.Foundation;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Input;
+
+namespace Uno.Material.Controls
+{
+	[TemplateVisualState(GroupName = __sheetStateGroup, Name = __collapsedState)]
+	[TemplateVisualState(GroupName = __sheetStateGroup, Name = __expandedState)]
+	public partial class ExpandingBottomSheet : Control
+	{
+		private const string __sheetStateGroup = "SheetStates";
+		private const string __collapsedState = "Collapsed";
+		private const string __expandedState = "Expanded";
+
+		public ExpandingBottomSheet()
+		{
+			DefaultStyleKey = typeof(ExpandingBottomSheet);
+		}
+
+		protected override void OnApplyTemplate()
+		{
+			if (IsEnabled)
+			{
+				VisualStateManager.GoToState(this, "Normal", true);
+			}
+			else
+			{
+				VisualStateManager.GoToState(this, "Disabled", true);
+			}
+
+			base.OnApplyTemplate();
+		}
+
+		private void UpdateVisualStates(bool useTransitions)
+		{
+			if (IsExpanded)
+			{
+				VisualStateManager.GoToState(this, __expandedState, useTransitions);
+			}
+			else
+			{
+				VisualStateManager.GoToState(this, __collapsedState, useTransitions);
+			}
+		}
+
+		private void ToggleSheet(object d, RoutedEventArgs e)
+		{
+			IsExpanded = !IsExpanded;
+		}
+
+		#region IsExpanded
+		public bool IsExpanded
+		{
+			get { return (bool)GetValue(IsExpandedProperty); }
+			set { SetValue(IsExpandedProperty, value); }
+		}
+
+		public static readonly DependencyProperty IsExpandedProperty =
+			DependencyProperty.Register("IsExpanded", typeof(bool), typeof(ExpandingBottomSheet), new PropertyMetadata(false, OnIsExpandedChanged));
+
+		private static void OnIsExpandedChanged(object d, DependencyPropertyChangedEventArgs e)
+		{
+			var sheet = d as ExpandingBottomSheet;
+			sheet?.UpdateVisualStates(true);
+		}
+		#endregion
+
+		#region CollapsedLabel, CollapsedTemplate and MinimalCollapsedTemplate
+		public object CollapsedLabel
```

Comment:
The way the property is used I would tend to think it should be CollapsedContent ?

I think it could be used as straight xaml like on a button when you set content inline. Or used as string along with a CollapsedTemplate